libs/auth-api-lib/src/lib/delegations/delegations-incoming.service.ts

```diff
@@ -224,6 +261,56 @@ export class DelegationsIncomingService {
     return [...mergedDelegationMap.values()]
   }
 
+  private async getAvailableDistrictCommissionersRegistryDelegations(
+    user: User,
+    types: AuthDelegationType[],
+    clientAllowedApiScopes: ApiScopeInfo[],
+    requireApiScopes?: boolean,
+  ): Promise<MergedDelegationDTO[]> {
+    const records =
+      await this.delegationsIndexService.getAvailableDistrictCommissionersRegistryRecords(
+        user,
+        types,
+        clientAllowedApiScopes,
+        requireApiScopes,
+      )
+    const merged = records.map((d) =>
+      DelegationDTOMapper.recordToMergedDelegationDTO(d),
+    )
+
+    const persons = (
+      await Promise.all(
+        merged.map((d) =>
+          this.nationalRegistryClient
+            .getIndividual(d.fromNationalId)
+            .catch((error) => error),
+        ),
+      )
+    )
+      .filter(this.isNotError)
+      .filter(isDefined)
+      .map((individual) => ({
+        nationalId: individual.nationalId,
+        name: individual.name ?? UNKNOWN_NAME,
+      }))
+
+    merged.forEach((d) => {
+      const person = persons.find((p) => p.nationalId === d.fromNationalId)
+      if (person) {
+        d.fromName = person.name
+      }
+    })
```

Comment:
Here you are looping through the `merged` array 5 times and the last times O(n^2) due to the looping of `persons`.

You should be able to just go through `merged` once by doing something like

```
const updateName = async (mergedDelegation: MergedDelegationDto): Promise<void> => new Promise<void>(async (resolve) => {
  let fromIndividual: IndividualDto | null
  
  try {
    fromIndividual = await this.nationalRegistryClient.getIndividual(mergedDelegation.fromNationalId))
  } catch (error) {
    // maybe log error
    fromIndividual = null
  }
  mergedDelegation.fromName = fromIndividual?.name ?? UNKNOWN_NAME

  resolve()
}

await Promise.all(
  merged.map((d) => updateName(d))
)
```

Reply:
Nice, updated.